botnim/vector_store/vector_store_es.py

```diff
@@ -79,135 +81,194 @@ def parse_index_name(index_name: str) -> str:
             return '', '', DEFAULT_ENVIRONMENT
         
 
-    def _build_search_query(self, query_text: str, embedding: List[float], 
-                          num_results: int = 7) -> Dict[str, Any]:
-        """Build the hybrid search query"""
-        text_match = {
-            "multi_match": {            # fields to search in
-                "query": query_text,
-                "fields": [
-                    "content",
-                    "metadata.title",
-                    "metadata.extracted_data.DocumentTitle",
-                    "metadata.extracted_data.DocumentTitle.keyword^3",  # Boost keyword matches
-                    "metadata.extracted_data.OfficialSource",
-                    "metadata.extracted_data.OfficialRoles.Role",
-                    "metadata.extracted_data.Description",
-                    "metadata.extracted_data.AdditionalKeywords",
-                    "metadata.extracted_data.Topics",
-                ],
-                "boost": 0.4,           # boost for text match vs vector match
-                "type": 'cross_fields',  # type of search: cross_fields, bool, simple, phrase, phrase_prefix
-                "operator": 'or',       # operator: or, and 
-            }
-        }
-        
-        # Enhanced priority for document title matches
-        # This creates several potential exact title matches by taking the first 1-3 words of the query
-        title_matches = []
-        words = query_text.split()
+    def _build_search_query(
+        self,
+        query_text: str,
+        search_mode: Optional[SearchModeConfig] = None,
+        embedding: Optional[List[float]] = None,
+        num_results: int = 7
+    ) -> Dict[str, Any]:
+        """
+        Builds an Elasticsearch query based on the search mode configuration.
         
-        # Try exact match with full query
-        title_matches.append({
-            "term": {
-                "metadata.extracted_data.DocumentTitle.keyword": {
-                    "value": query_text,
-                    "boost": 10.0
+        Args:
+            query_text: The search query string
+            search_mode: Optional search mode configuration
+            embedding: Optional embedding vector for vector search
+            num_results: Number of results to return
+            
+        Returns:
+            Dict containing the Elasticsearch query
+        """
+        if not search_mode:
```

Comment:
It would have been much better to be able to generate the 'default mode' query out of a default SearchMode object and have one code branch here instead of these two.

Reply:
done.